src/Runtime/LambdaRuntime.php

```diff
@@ -104,14 +109,23 @@ private function waitNextInvocation(): array
         }
 
         // Retrieve invocation ID
-        $invocationId = '';
-        curl_setopt($this->handler, CURLOPT_HEADERFUNCTION, function ($ch, $header) use (&$invocationId) {
+        $contextBuilder = new ContextBuilder;
+        curl_setopt($this->handler, CURLOPT_HEADERFUNCTION, function ($ch, $header) use ($contextBuilder) {
             if (! preg_match('/:\s*/', $header)) {
                 return strlen($header);
             }
             [$name, $value] = preg_split('/:\s*/', $header, 2);
             if (strtolower($name) === 'lambda-runtime-aws-request-id') {
-                $invocationId = trim($value);
+                $contextBuilder->setAwsRequestId(trim($value));
```

Comment:
Maybe do a switch/case to prevent unnecessary condition call or if/elseif.

if you don't do that, at least, resolve strtolower($name) and trim($value) just one time.

Reply:
they're not nested ifs but the strtolower/trim is good catch. too much copy/pasta going on